runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -314,21 +228,63 @@ protected void markTile(LocalPoint localPoint)
 		WorldPoint worldPoint = WorldPoint.fromLocalInstance(client, localPoint);
 
 		int regionId = worldPoint.getRegionID();
-		GroundMarkerPoint point = new GroundMarkerPoint(regionId, worldPoint.getX() & 0x3f, worldPoint.getY() & 0x3f, client.getPlane());
-		log.debug("Updating point: {} - {}", point, worldPoint);
+		GroundMarkerPoint marker = new GroundMarkerPoint(regionId, worldPoint.getX() & 0x3f, worldPoint.getY() & 0x3f, client.getPlane(), config.markerColor());
+		log.debug("Updating point: {} - {}", marker, worldPoint);
 
-		List<GroundMarkerPoint> points = new ArrayList<>(getPoints(regionId));
-		if (points.contains(point))
+		List<GroundMarkerPoint> groundMarkers = new ArrayList<>(getPoints(regionId));
+		if (groundMarkers.contains(marker))
 		{
-			points.remove(point);
+			groundMarkers.remove(marker);
 		}
 		else
 		{
-			points.add(point);
+			groundMarkers = groundMarkers.stream().filter(m -> !m.sameTile(marker)).collect(Collectors.toList());
+			groundMarkers.add(marker);
 		}
 
-		savePoints(regionId, points);
+		savePoints(regionId, groundMarkers);
 
 		loadPoints();
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Updates configuration data for ground markers to store color information.
+	 * This method should be removed after a reasonable amount of time.
+	 */
+	@Deprecated
+	void setColorConfiguration()
```

Comment:
I don't see any need for this at all. If it has no color then it has no color, so use whatever is configured.

Reply:
My thought process behind this is that that this PR removes the ability to have existing markers change upon configured plugin color changing, so it would be kind of weird for markers existing before this is merged to change when any new ones would be of a fixed color. I can change it like you requested anyway, but it just seems weird to introduce inconsistent behavior that way.

Reply:
Imo this is not necessary too, or it can be done in better way? E.g on point loading, check if point color is null and if yes, set it to current color and save. Would be a lot simpler than this "abomination" 😄

Reply:
I got myself confused on separating a private encapsulation of `WorldPoint` and `Color` and thought at the time that migrating configs to use worldpoint would be the best solution there. Your suggestion makes sense, and that's what I've done in my recent push--serialization is no longer converted and plugin logic now uses the new `ColorTileMarker` encapsulation for rendering.